core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts

```diff
@@ -82,121 +82,30 @@ export class DotEmaShellComponent implements OnInit, OnDestroy {
-                        isDisabled: !page.canEdit
-                    },
-                    {
-                        iconURL: 'experiments',
-                        label: 'editema.editor.navbar.experiments',
-                        href: `experiments/${page.identifier}`,
-                        isDisabled: !page.canEdit
-                    },
-                    {
-                        icon: 'pi-th-large',
-                        label: 'editema.editor.navbar.page-tools',
-                        action: () => {
-                            this.pageTools.toggleDialog();
-                        }
-                    },
-                    {
-                        icon: 'pi-ellipsis-v',
-                        label: 'editema.editor.navbar.properties',
-                        action: () => {
-                            this.dialog.editContentlet({
-                                inode: page.inode,
-                                title: page.title,
-                                identifier: page.identifier,
-                                contentType: page.contentType
-                            });
-                        }
-                    }
-                ],
-                canRead: page.canRead,
-                seoProperties: {
-                    currentUrl,
-                    languageId,
-                    siteId,
-                    requestHostName: host
-                },
-                error
-            };
-        })
-    );
     readonly #activatedRoute = inject(ActivatedRoute);
     readonly #router = inject(Router);
     readonly #siteService = inject(SiteService);
     readonly #dotMessageService = inject(DotMessageService);
     readonly #confirmationService = inject(ConfirmationService);
 
+    protected readonly $shellProps = this.uveStore.$shellProps;
+
     readonly #destroy$ = new Subject<boolean>();
     #currentComponent: unknown;
 
-    // We can internally navigate, so the PageID can change
+    readonly translatePageEffect = effect(() => {
+        const { languages, languageId, page } = this.uveStore.$shellProps().translateProps;
 
-    get queryParams(): DotPageApiParams {
-        const queryParams = this.#activatedRoute.snapshot.queryParams;
+        if (languages.length) {
+            const currentLanguage = languages.find((lang) => lang.id === languageId);
 
-        return {
-            language_id: queryParams['language_id'],
-            url: queryParams['url'],
-            'com.dotmarketing.persona.id': queryParams['com.dotmarketing.persona.id'],
-            variantName: queryParams['variantName'],
-            clientHost: queryParams['clientHost']
-        };
-    }
+            if (!currentLanguage.translated) {
+                this.createNewTranslation(currentLanguage, page);
+            }
+        }
+    });
 
     ngOnInit(): void {
         combineLatest([this.#activatedRoute.data, this.#activatedRoute.queryParams])
```

Comment:
@zJaaal It's possible to avoid handle the manual destroy using `takeUntilDestroyed` from `@angular/core/rxjs-interop`, example:

`.pipe(takeUntilDestroyed())`

this method automatically handles the proper context and destroy